libraries/redcore/api/hal/hal.php

```diff
@@ -1376,16 +1388,19 @@ public function checkRequiredFields($data, $configuration)
 				{
 					if (is_null($data[(string) $field['name']]) || $data[(string) $field['name']] == '')
 					{
-						JFactory::getApplication()->enqueueMessage(
-							JText::sprintf('LIB_REDCORE_API_HAL_WEBSERVICE_ERROR_REQUIRED_FIELD', (string) $field['name']), 'error'
-						);
-
-						return false;
+						$errors[] = JText::sprintf('LIB_REDCORE_API_HAL_WEBSERVICE_ERROR_REQUIRED_FIELD', (string) $field['name']);
 					}
 				}
 			}
 		}
 
+		if (!empty($errors))
+		{
+			$this->apiErrors = array_merge($this->apiErrors, $errors);
```

Comment:
the variable is here https://github.com/tharunaredweb/redCORE/blob/4df7b3a3fbe860eaafeaa9c192ed74664f78d616/libraries/redcore/api/hal/hal.php#L1393